api/src/pcapi/core/users/api.py

```diff
@@ -1558,10 +1561,58 @@ def anonymize_non_pro_non_beneficiary_users(*, force: bool = False) -> None:
             ~users_models.User.email.like("%@passculture.app"),  # people who work or worked in the company
             func.array_length(users_models.User.roles, 1).is_(None),  # no role, not already anonymized
             finance_models.Deposit.userId.is_(None),  # no deposit
-            users_models.User.lastConnectionDate < datetime.datetime.utcnow() - datetime.timedelta(days=3 * 365),
+            users_models.User.lastConnectionDate < datetime.datetime.utcnow() - relativedelta(years=3),
+        )
+        .all()
+    )
+    for user in users:
+        anonymize_user(user, force=force)
+    db.session.commit()
+
+
+def anonymize_beneficiary_users(*, force: bool = False) -> None:
+    """
+    Anonymize user accounts that have been beneficiaries which have not connected for at least 3 years, and
+    whose deposit has been expired for at least 5 years.
+    """
+    users = (
+        users_models.User.query.outerjoin(
+            finance_models.Deposit,
+            users_models.User.deposits,
+        )
+        .filter(
+            users_models.User.is_beneficiary.is_(True),  # type: ignore [attr-defined]
+            sa.and_(
+                users_models.User.lastConnectionDate < datetime.datetime.utcnow() - relativedelta(years=3),
+                finance_models.Deposit.expirationDate < datetime.datetime.utcnow() - relativedelta(years=5),
+            ),
         )
         .all()
     )
     for user in users:
         anonymize_user(user, force=force)
     db.session.commit()
+
+
+def anonymize_user_deposits() -> None:
+    """
+    Anonymize deposits that have been expired for at least 10 years.
+    """
+    deposits = finance_models.Deposit.query.filter(
```

Comment:
Juste une micro-remarque question de goût personnel : `deposits` me suggère qu'on récupère les objets, or ici ce n'est plus le résultat du `.all()` :
```suggestion
    deposits_query = finance_models.Deposit.query.filter(
```